src/Symfony/Component/Intl/DateFormatter/IntlDateFormatter.php

```diff
@@ -132,7 +132,7 @@ class IntlDateFormatter
      * @throws MethodArgumentValueNotImplementedException When $locale different than "en" or null is passed
      * @throws MethodArgumentValueNotImplementedException When $calendar different than GREGORIAN is passed
      */
-    public function __construct(?string $locale, int $datetype, int $timetype, $timezone = null, ?int $calendar = self::GREGORIAN, string $pattern = null)
+    public function __construct(?string $locale, int $datetype = null, int $timetype = null, $timezone = null, ?int $calendar = self::GREGORIAN, string $pattern = null)
```

Comment:
Shouldn't it be this since the arguments are required, but null can be passed?
```suggestion
    public function __construct(?string $locale, ?int $datetype, ?int $timetype, $timezone = null, ?int $calendar = self::GREGORIAN, string $pattern = null)
```

Reply:
no, we always write it like done here

Reply:
What we do is never right `?int $datetype = null` but instead use `int $datetype = null` because  they work exactly the same.

But the change I proposed it to completely remove the default value because that's actualy how the native PHP class works: you need to pass at least 3 arguments when creating an `IntlDateFormatter`, but they can be `null`.

Reply:
Gotcha, I missed that you cannot call the native's class constructor with fewer arguments.